File: app/src/main/java/ru/merkulyevsasha/news/data/prefs/NewsSharedPreferencesImpl.java
package ru.merkulyevsasha.news.data.prefs;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;

import io.reactivex.Single;

/**
 * Created by sasha_merkulev on 21.09.2017.
 */

public class NewsSharedPreferencesImpl implements NewsSharedPreferences {

    private final static String SHARED_PREF_NAME = "news_pref";

    private final static String KEY_FIRST_RUN = "ru.merkulyevsasha.news.prefs_key_run_flag";

    private final SharedPreferences prefs;

    @Inject
    public NewsSharedPreferencesImpl(Context context){
        prefs = context.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE);
    }

    @Override
    public Single<Boolean> getFirstRunFlag() {
        return Single.fromCallable(() -> prefs.getBoolean(KEY_FIRST_RUN, true));
    }

    @Override
    public void setFirstRunFlag() {
        SharedPreferences.Editor editor = prefs.edit();
        editor.putBoolean(KEY_FIRST_RUN, false);
        editor.apply();
    }

}

File: app/src/main/java/ru/merkulyevsasha/news/data/prefs/NewsSharedPreferencesImpl.kt
package ru.merkulyevsasha.news.data.prefs

import android.content.Context
import android.content.SharedPreferences

import javax.inject.Inject

import io.reactivex.Single

class NewsSharedPreferencesImpl
@Inject constructor(context: Context) : NewsSharedPreferences {

    private val prefs: SharedPreferences = context.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)

    override fun getFirstRun(): Single<Boolean> {
        return Single.fromCallable { prefs.getBoolean(KEY_FIRST_RUN, true) }
    }

    override fun setFirstRun() {
        prefs.edit()
            .putBoolean(KEY_FIRST_RUN, false)
            .apply()
    }

    override fun getProgress(): Single<Boolean> {
        return Single.fromCallable { prefs.getBoolean(KEY_PROGRESS, false) }
    }

    override fun setProgress(progress: Boolean) {
        prefs.edit()
            .putBoolean(KEY_PROGRESS, progress)
            .apply()
    }

    companion object {
        private const val SHARED_PREF_NAME = "news_pref"
        private const val KEY_FIRST_RUN = "prefs_key_run_flag"
        private const val KEY_PROGRESS = "prefs_key_progress_flag"
    }

}
